Automattic/newspack-theme
# Remove escaping from custom colour and typography CSS

### All Submissions:


### Changes proposed in this Pull Request:

I noticed in working a the 'Style 2' child theme PR that the `esc_html()` function used to escape the Custom Colour CSS in the child themes was turning `>` to `&gt;`, breaking some of the CSS selectors. `wp_kses()`, used by the Typography CSS, is doing the same thing.

Since the Colour and Typography CSS is not user-generated, but is hard-coded strings passed from a different file, they don't actually need escaping. So I've removed the functions, and marked the functions as being 'safe'. 

**Edited to add:** This PR has been updated to escape custom colour and typography variables in the color-patterns and typography files, and to remove the unused colour filter code (more details in [this comment](https://github.com/Automattic/newspack-theme/pull/698#issuecomment-575227414)).

### How to test the changes in this Pull Request:

1. Apply the PR.
2. Try switching the custom colours and type; ensure you don't see any errors.
3. Try switching the theme to 'Newspack Sacha', and repeat step 2.
4. Try switching the theme to 'Newspack Scott', and repeat step 2.
5. Review the changes in the color-patterns.php and child-color-patterns.php files; ensure each outputted colour variable is escaped with `esc_html()`, and that there are no typos.
6. Review the changes in the typography.php and child-typography.php files; ensure each outputted typography variable is escaped with `wp_kses()`, with the allowed tags value of `null`.
7. Check the editor and make sure the colours populating the palette match your current colour settings.

### Other information:
